exporters/metric/cortex/cortex.go

```diff
@@ -85,6 +93,231 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name := sanitize(record.Descriptor().Name())
+	labels := createLabelSet(record, "name", name)
+
+	// Create TimeSeries and return
+	tSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{sample},
+		Labels:  labels,
+	}
+
+	return tSeries, nil
+}
+
+// convertFromMinMaxSumCount returns 4 TimeSeries for the min, max, sum, and count from the mmsc aggregation
+func convertFromMinMaxSumCount(record metric.Record, minMaxSumCount aggregation.MinMaxSumCount) ([]*prompb.TimeSeries, error) {
+	// Convert Min
+	min, err := minMaxSumCount.Min()
+	if err != nil {
+		return nil, err
+	}
+	minSample := prompb.Sample{
+		Value:     min.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name := sanitize(record.Descriptor().Name() + "_min")
+	labels := createLabelSet(record, "name", name)
+
+	// Create TimeSeries
+	minTimeSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{minSample},
+		Labels:  labels,
+	}
+
+	// Convert Max
+	max, err := minMaxSumCount.Max()
+	if err != nil {
+		return nil, err
+	}
+	maxSample := prompb.Sample{
+		Value:     max.CoerceToFloat64(record.Descriptor().NumberKind()),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name = sanitize(record.Descriptor().Name() + "_max")
+	labels = createLabelSet(record, "name", name)
+
+	// Create TimeSeries
+	maxTimeSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{maxSample},
+		Labels:  labels,
+	}
+
+	// Convert Count
+	count, err := minMaxSumCount.Count()
+	if err != nil {
+		return nil, err
+	}
+	countSample := prompb.Sample{
+		Value:     float64(count),
+		Timestamp: record.EndTime().Unix(), // Convert time to Unix (int64)
+	}
+
+	// Create labels, including metric name
+	name = sanitize(record.Descriptor().Name() + "_count")
+	labels = createLabelSet(record, "name", name)
+
+	// Create TimeSeries
+	countTimeSeries := &prompb.TimeSeries{
+		Samples: []prompb.Sample{countSample},
+		Labels:  labels,
+	}
+
+	tSeries := []*prompb.TimeSeries{
+		minTimeSeries, maxTimeSeries, countTimeSeries,
+	}
+
+	return tSeries, nil
+}
+
+// createLabelSet combines labels from a Record, resource, and extra labels to
+// create a slice of prompb.Label
+func createLabelSet(record metric.Record, extras ...string) []*prompb.Label {
+	// Map ensure no duplicate label names
+	labelMap := map[string]prompb.Label{}
+
+	// mergeLabels merges Record and Resource labels into a single set, giving
+	// precedence to the record's labels.
+	mi := label.NewMergeIterator(record.Labels(), record.Resource().LabelSet())
+	for mi.Next() {
+		label := mi.Label()
+		key := string(label.Key)
+		labelMap[key] = prompb.Label{
+			Name:  sanitize(key),
```

Comment:
I added a check in this method where if the label name starts with__ and ends with __, it won't get sanitized